snap/snapcraft.yaml

```diff
@@ -0,0 +1,57 @@
+name: gdc
+version: 7.3.0
+summary: D compiler based on the GCC backend
+description: |
+    GDC is a compiler for the D programming language, which
+    integrates the official DMD D2 compiler frontend with
+    GCC.
+
+confinement: classic
+grade: devel
+
+apps:
+  gdc:
+    command: bin/gdc
+  gdmd:
+    command: bin/gdmd
+
+parts:
+  gdc:
+    source: https://github.com/D-Programming-GDC/GDC.git
+    source-branch: gdc-7
+    plugin: dump
+    stage:
+    - -*
+    prime:
+    - -*
+
+  gdmd:
+    source: https://github.com/D-Programming-GDC/GDMD.git
+    source-branch: master
+    plugin: dump
+    organize:
+      dmd-script: bin/gdmd
+      dmd-script.1: man/man1/gdmd.1
+
+  gcc:
+    source: ftp://ftp.mirrorservice.org/sites/sourceware.org/pub/gcc/releases/gcc-7.3.0/gcc-7.3.0.tar.gz
+    source-checksum: sha512/4e203f4ea5e8713e7b0e3d2a269f7a54f6d1074d572b93d39ed6961c82b3c310f389d7f78494f58309b7436d1e0744eba06c22a24747000dfd84e2b4376cbf73
+    plugin: autotools
+    configflags:
+    - --enable-languages=d,lto
+    - --enable-checking=release
+    - --disable-bootstrap
```

Comment:
Or the baseline arch?

Reply:
PIE should be turned on, yes.  What do you mean by "the baseline arch"?

Ideally I'd like the package to provide as many possible build targets as possible.

Reply:
Each architecture (in Debian/Ubuntu at least) is usually configured with a specific target triplet that is non-default for vanilla builds.  Mostly to drop support for older CPUs in order to take advantage of more optimal instructions.

For instance, cpu=i686 instead of i386, or ARMv7 instead of ARMv5, or MIPS II instead of MIPS I, POWER7 instead of POWER5. List goes on... ☺️

Reply:
Well, I think it's probably best to not be too smartarse for a first iteration.  What do I need to select to enable PIE?  Other than that, let's go with the defaults.

Oh, and let's touch base tomorrow to talk about actually setting up the snap package in the snap store, we'll probably need to make an application to get the `gdc` name.

Reply:
Sure.  I do wonder however how they manage the gcc snap package, assuming one exists.

Reply:
I don't think there is one.